openvas fails to compile on clang 15+

<!--
If reporting an issue please try to provide the information asked below.

Before reporting an issue please:

1. Be aware that this is not a support forum. If your issue is rather a question
   than a bug report, please use our community forum at
   https://community.greenbone.net/c/gse instead.
2. Make sure that you're using the latest published GVM components for the
   release you are using: https://community.greenbone.net/t/about-the-source-edition-gse-category/176
3. Check the list of issues whether it isn't already reported.
4. Read 1. again and if you still believe you found a software bug please
   continue to file this issue. If you are in doubt use
   https://community.greenbone.net/c/gse instead.

Thanks for your help to keep the communication channels clean and consistent!
-->

despite fixes in https://github.com/greenbone/openvas-scanner/pull/1025 clang 15+ still reports issues. this was originally reported as: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=270464

### Expected behavior

Build succeeds

### Actual behavior

build fails:

```
misc/CMakeFiles/openvas_misc_shared.dir/plugutils.c.o -MF misc/CMakeFiles/openvas_misc_shared.dir/plugutils.c.o.d -o misc/CMakeFiles/openvas_misc_shared.dir/plugutils.c.o -c /wrkdirs/usr/ports/security/openvas/work/openvas-scanner-22.4.0/misc/plugutils.c
/wrkdirs/usr/ports/security/openvas/work/openvas-scanner-22.4.0/misc/plugutils.c:841:22: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
plug_get_key_sigchld ()
                     ^
                      void
1 error generated. 
```

### Steps to reproduce

1. install llvm 15+
2. attempt to build
3. build fails


### Environment

**Operating system:**

see https://portsfallout.com/fallout?port=security%2Fopenvas%24 for all the FreeBSD versions where this fails

--- a/misc/plugutils.c
+++ b/misc/plugutils.c
@@ -1072,9 +1072,10 @@ plug_get_kb (struct script_infos *args)
 }
 
 static void
-plug_get_key_sigchld ()
+plug_get_key_sigchld (int s)
 {
   int status;
+  (void) s;
 
   wait (&status);
 }

